tests/python/test_logging.py

```diff
@@ -0,0 +1,45 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2022, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from smartredis import *
+from smartredis.error import *
+
+
+def test_logging(use_cluster, context):
```

Comment:
incredibly at risk of over complicating the test suite, but very similar to my previous note, you could do

```py
@pytest.mark.parameterize("log_fn, log_level", [
    itertools.product(
        [log_data, log_warning, log_error],
        [LLQuiet, LLInfo, LLDebug, LLDeveloper])
])
def test_logging(use_cluster, context, log_fn, log_level):
    ...
```
So that if one of the cases below were to fail, the rest would still run

Reply:
I was able to use this idea partially to streamline the test by parameterizing over the log levels. The differences in the log messages make parameterizing over log_fn add more code than it saves